aggregator.py

```diff
@@ -497,6 +502,47 @@ def parse_event_packet(self, packet):
         except (IndexError, ValueError):
             raise Exception(u'Unparseable event packet: %s' % packet)
 
+    def parse_sc_packet(self, packet):
+        try:
+            _, data_and_metadata = packet.split('|', 1)
+            # Service check syntax:
+            # _sc|check_name|status|meta
+            if data_and_metadata.count('|') == 1:
+                # Case with no metadata
+                check_name, status = data_and_metadata.split('|')
+                metadata = ''
+            else:
+                check_name, status, metadata = data_and_metadata.split('|', 2)
+
+            service_check = {
+                'check_name': check_name,
+                'status': int(status)
+            }
+
+            message_delimiter = '|m:' if '|m:' in metadata else 'm:'
+            if message_delimiter in metadata:
+                meta, message = metadata.rsplit(message_delimiter, 1)
+                service_check['message'] = self._unescape_sc_content(message)
+            else:
+                meta = metadata
+
+            if not meta:
+                return service_check
+
+            meta = unicode(meta)
```

Comment:
That wouldn't work with UTF-8 encoded packets. Could you have a look into this PR https://github.com/DataDog/dd-agent/pull/1279 and mimic the behavior ? 
cc: @LeoCavaille